src/index.js

```diff
@@ -1,3 +1,137 @@
+        if (item.completed) li.querySelector(".toggle").setAttribute("checked", item.completed);
+        li.querySelector(".destroy").addEventListener("click", onClickDestroy);
+
+        li.addEventListener('dblclick',onDoubleClickTodoItem);
+        todoList.append(li);
+    });
+    document.querySelector("#new-todo-title").value = '';
+    document.querySelector(".todo-count strong").textContent = state.items.length;
+    localStorage.setItem("state", JSON.stringify(state));
+}
+
+// Events
+// 이벤트들은 전역객체 상태값을 변경하고, 함수의 마지막 부분에서 rePaint 함수를 실행시킨다.
+function onChangeCheckBox(e) {
+    const li = e.target.parentNode.parentNode;
+    const targetId = li.dataset.id;
+    const origin = state.items[targetId];
+    origin.completed = e.target.checked;
+    rePaint(state);
+}
+function onClickDestroy(e) {
+    const li = e.target.parentNode.parentNode;
+    const targetId = li.dataset.id;
+    state.items.splice(targetId,1);
+    rePaint(state);
+}
+function onDoubleClickTodoItem(e) {
+    const li = e.target.parentNode.parentNode;
+    const targetId = li.dataset.id;
+    const origin = state.items[targetId];
+    origin.editing = true;
+    // li.querySelector(".edit").focus();
+    rePaint(state);
+}
+function onEdit(e) {
+    const li = e.target.parentNode;
+    const targetId = li.dataset.id;
+    if(e.keyCode === 27) {
+        state.items[targetId].editing = false;
+        rePaint(state);
+    }
+    if(e.keyCode === 13) {
+        state.items[targetId] = {...state.items[targetId], todo: e.target.value, editing: false};
+        rePaint(state);
+    }
+}
+function onKeyUpEnter(e) {
+    if(e.keyCode === 13) {
+        const item = {
+            todo: e.target.value,
+            completed: false,
+            editing: false
+        }
+        state.items = [...state.items, item];
+        rePaint(state);
+    }
+}
+
+// 초기화 함수
+// todoItem들을 제외한 요소들에 event를 등록한다.
+// localStorage에 기존에 저장된 상탯값이 있다면 그 것으로 state를 초기화한다.
+// 함수의 마지막에 rePaint를 실행한다.
+function Initialize() {
+    const localState = JSON.parse(localStorage.getItem("state"));
+    if (localState) {
+        state = {
+            ...localState
+        };
+    }
+    const todoInput = document.getElementById("new-todo-title");
+    todoInput.addEventListener("keyup", onKeyUpEnter);
+    const activeConditionBtn = document.querySelector("a.active");
+    const completedConditionBtn = document.querySelector("a.completed");
+    const allConditionBtn = document.querySelector("a.all");
+    allConditionBtn.addEventListener("click", (e)=>{
+        e.preventDefault();
+        state.condition = "all";
+        rePaint(state);
+    });
+    activeConditionBtn.addEventListener("click", (e)=>{
+        e.preventDefault();
+        state.condition = "active";
+        rePaint(state);
+    });
+    completedConditionBtn.addEventListener("click",(e)=>{
+        e.preventDefault();
+        state.condition = "completed";
+        rePaint(state);
+    });
```

Comment:
```suggestion
  const setCondition = ({$target, condition}) => {
    $target.addEventListener('click', (event) => {
      event.preventDefault();
      state.condition = condition;
      rePaint(state);
    })
  };

  setCondition({ $target: allConditionBtn, condition: 'all'});
  setCondition({ $target: activeConditionBtn, condition: 'active'});
  setCondition({ $target: completedConditionBtn, condition: 'completed'});
```
이런식으로 바꿔보는 건 어떨까요? 

filter 버튼이 a 태그인지 미처 생각하지 못했는데 `e.preventDefault();` 를 보고 알게되었습니다!

그리고 condition 에 따라 아이템들이 filter 는 잘 동작이 되고 있습니다. 하지만 condition 이 바뀌었을 때 선택된 버튼의 색깔이 바뀌는 기능에 대해서 추가적으로 작성을 하시면 좋을 것 같습니다